learing_journal/test_security.py

```diff
@@ -0,0 +1,151 @@
+'''
+    These tests where typed out during lecture
+    from Cris Ewing September 1, 2016
+'''
+import os
+from pyramid import testing
+
+
+def test_public_view(app):
+    '''tests that the public view is public'''
+    response = app.get('/public')
+    assert response.status_code == 200
+
+
+def test_private_view_Inaccessable(app):
+    '''tests that the priveate view is private'''
+    response = app.get('/private', status='4*')
+    # import pdb; pdb.set_trace()
+    assert response.status_code == 403
+
+
+def test_private_view_accaessible_to_authenticated(authenticated_app):
+    '''test that the private view is accessable to authenticated'''
+    response = authenticated_app.get('/private', status='2*')
+    assert response.status_code == 200
+
+
+def test_create_view_inaccessable(app):
+    '''test the create view is inaccessable to nonauthorized'''
+    response = app.get('/journal/new-entry', status='4*')
+    assert response.status_code == 403
+
+
+def test_create_view_accessible_to_authenticated(authenticated_app):
+    '''ttest that the create view is accessable to authenticated user'''
+    response = authenticated_app.get('/journal/new-entry', status='2*')
+    assert response.status_code == 200
+
+
+def test_update_view_inaccessable(app):
+    '''test that the update view is inaccesable to unauthorized user'''
+    response = app.get('/journal/1/edit-entry', status='4*')
+    assert response.status_code == 403
+
+
+def test_update_view_accessible_to_authenticated(authenticated_app):
+    '''test the update view is accessable to authenticated user'''
+    response = authenticated_app.get('/journal/1/edit-entry', status='2*')
+    assert response.status_code == 200
+
+
+def test_detail_view_inaccessable(app):
+    '''test the detail view is inaccessable to unauthorized'''
+    response = app.get('/journal/1', status='4*')
+    assert response.status_code == 403
```

Comment:
Is this really what you want?  An unauthenticated user should not be able to read your journal entries?  How am I going to read them when grading your learning journal submissions?